Base/QTGUI/qSlicerApplication.cxx

```diff
@@ -178,9 +178,11 @@ qSlicerApplicationPrivate::qSlicerApplicationPrivate(
 qSlicerApplicationPrivate::~qSlicerApplicationPrivate()
 {
   // Delete settings dialog. deleteLater would cause memory leaks on exit.
-  this->SettingsDialog->setParent(nullptr);
-  delete this->SettingsDialog;
-  this->SettingsDialog = nullptr;
+  if (this->SettingsDialog)
+    {
+    this->SettingsDialog->setParent(nullptr);
```

Comment:
Describe here why we must choose check if non-null (because the main window may have already deleted it) and why we need to set the parent to null (because it may have been assigned to the main window).

Reply:
fixed